sys/debug/kgmon.py

```diff
@@ -17,7 +18,33 @@ def gmon_write(path):
         kcount = gdb.parse_and_eval('_gmonparam.kcount')
         of.write(infer.read_memory(kcount, kcountsize))
 
-        # TODO: write arc info
+        # Write arc info
+        froms_el_size = int(gdb.parse_and_eval('sizeof(*_gmonparam.froms)'))
+        fromssize = int(gdb.parse_and_eval('_gmonparam.fromssize'))
+        tossize = int(gdb.parse_and_eval('_gmonparam.tossize'))
+        hashfraction = gdb.parse_and_eval('_gmonparam.hashfraction')
+        froms = gdb.parse_and_eval('_gmonparam.froms')
+        tos = gdb.parse_and_eval('_gmonparam.tos')
+        lowpc = gdb.parse_and_eval('_gmonparam.lowpc')
+
+        memory = infer.read_memory(froms, fromssize)
+        froms_array = unpack('H' * int(fromssize/calcsize('H')), memory)
+        memory = infer.read_memory(tos, tossize)
+        tos_array = unpack('IiHH' * int(tossize/calcsize('IiHH')), memory)
+
+        fromindex = 0
+        for from_val in froms_array:
+            if from_val == 0:
+                continue
+            frompc = lowpc + fromindex * froms_el_size * hashfraction
+            toindex = from_val
+
+            while toindex != 0:
+                selfpc = tos_array[toindex * 4]
+                count = tos_array[toindex * 4 + 1]
+                toindex = tos_array[toindex * 4 + 2]
+                of.write(pack('IIi', frompc, selfpc, count))
+            fromindex += 1
```

Comment:
A little comment about layout and meaning of these values would be helpful.

Reply:
Done, let me know if it is enough?